Chat/Network.cpp

```diff
@@ -261,8 +256,57 @@ void Network::ProcessMessage(const RecvStruct &recv_str)
 
 void Network::GetOnlineUsers(std::vector<std::string> &out_users)
 {
-    for (std::pair<std::string, std::string> pair_ : user_ip_name_map_)
-    {
-        out_users.push_back(pair_.second);
-    }
+    ip_name_list_.GetNameList(out_users);
+}
+
+unsigned Network::SendFileStartup(void *send_file_info)
+{
+    threads_num_mutex_.Lock();
+    file_sharing_thread_num_++;
+    threads_num_mutex_.Unlock();
+
+    FileSendSocket().SendFile((*(SendFileInfo*)send_file_info).path_,
+                              (*(SendFileInfo*)send_file_info).ip_,
+                              (*(SendFileInfo*)send_file_info).name_);
+    delete send_file_info;
+
+    threads_num_mutex_.Lock();
+    file_sharing_thread_num_--;
+    threads_num_mutex_.Unlock();
+
+    return 0;
+}
+
+void Network::SendFile(const std::string &path,
+                       const std::string &ip,
+                       const std::string &name)
+{
+    SendFileInfo *sfi = new SendFileInfo({ path, name, ip });
```

Comment:
implicit conversation to SendFileInfo
same thing is i wrote
SendFileInfo fi = {path, name, ip}
SendFileInfo *pFi = new SendFileInfo(fi); //copying